nifi-nar-bundles/nifi-hazelcast-bundle/nifi-hazelcast-services/src/main/java/org/apache/nifi/hazelcast/services/cachemanager/EmbeddedHazelcastCacheManager.java

```diff
@@ -0,0 +1,244 @@
+
+        // This validation also prevents early activation: up to the point node is considered as clustered this validation step
+        // prevents automatic enabling when the flow stars (and the last known status of the controller service was enabled).
+        if (!getNodeTypeProvider().isClustered() && context.getProperty(HAZELCAST_CLUSTERING_STRATEGY).getValue().equals(CLUSTER_ALL_NODES.getValue())) {
+            results.add(new ValidationResult.Builder()
+                    .subject(HAZELCAST_CLUSTERING_STRATEGY.getName())
+                    .valid(false)
+                    .explanation("Cannot use \"All Nodes\" Clustering Strategy when NiFi is not part of a cluster!")
+                    .build());
+        }
+
+        if (!getNodeTypeProvider().isClustered() && context.getProperty(HAZELCAST_CLUSTERING_STRATEGY).getValue().equals(CLUSTER_EXPLICIT.getValue())) {
+            results.add(new ValidationResult.Builder()
+                    .subject(HAZELCAST_CLUSTERING_STRATEGY.getName())
+                    .valid(false)
+                    .explanation("Cannot use \"Explicit\" Clustering Strategy when NiFi is not part of a cluster!")
+                    .build());
+        }
+
+        if (!context.getProperty(HAZELCAST_INSTANCES).isSet() && context.getProperty(HAZELCAST_CLUSTERING_STRATEGY).getValue().equals(CLUSTER_EXPLICIT.getValue())) {
+            results.add(new ValidationResult.Builder()
+                    .subject(HAZELCAST_INSTANCES.getName())
+                    .valid(false)
+                    .explanation("In case of \"Explicit\" Clustering Strategy, instances needs to be specified!")
+                    .build());
+        }
+
+        if (context.getProperty(HAZELCAST_INSTANCES).isSet() && !context.getProperty(HAZELCAST_CLUSTERING_STRATEGY).getValue().equals(CLUSTER_EXPLICIT.getValue())) {
+            results.add(new ValidationResult.Builder()
+                    .subject(HAZELCAST_INSTANCES.getName())
+                    .valid(false)
+                    .explanation("In case of other Clustering Strategy than \"Explicit\", instances should not be specified!")
+                    .build());
+        }
+
+        if (context.getProperty(HAZELCAST_INSTANCES).isSet() && context.getProperty(HAZELCAST_CLUSTERING_STRATEGY).getValue().equals(CLUSTER_EXPLICIT.getValue())) {
+            final Collection<String> niFiHosts = getNodeTypeProvider().getClusterMembers();
+            final Collection<String> hazelcastHosts = getHazelcastMemberHosts(context);
+
+            for (final String hazelcastHost : hazelcastHosts) {
+                if (!niFiHosts.contains(hazelcastHost)) {
+                    results.add(new ValidationResult.Builder()
+                            .subject(HAZELCAST_INSTANCES.getName())
+                            .valid(false)
+                            .explanation("Host \"" + hazelcastHost + "\" is not part of the NiFi cluster!")
```

Comment:
Same as the comment in line 207.

Reply:
The same applies: the list should help to keep the order of messages making it easier to understand.